addons/survey/static/src/js/survey_timer.js

```diff
@@ -1,45 +1,71 @@
 odoo.define('survey.timer', function (require) {
 'use strict';
 
-require('web.dom_ready');
+var publicWidget = require('web.public.widget');
 
-if (!$('.js_survey_timer').length) {
-    return Promise.reject("DOM doesn't contain '.js_survey_timer'");
-}
+var interval = null;
+var $timer = null;
+var countDownDate = null;
 
-var $parent = $('.js_survey_timer');
-var timeLimitMinutes = parseInt($parent.data('time_limit_minutes'));
+publicWidget.registry.SurveyTimerWidget = publicWidget.Widget.extend({
+    template: false,
+    events: {},
 
-if (timeLimitMinutes <= 0) {
-    return Promise.reject("Timer is not positive");
-}
+    //--------------------------------------------------------------------------
+    // Widget
+    //--------------------------------------------------------------------------
 
-var countDownDate = moment.utc($parent.data('timer')).add(timeLimitMinutes, 'minutes');
+    /**
+    * @override
+    */
+    start: function () {
+        // Validation
+        if (!$('.o_survey_timer').length) {
```

Comment:
Don't need this anymore since the widget is directly instantiated with that very dom element.